spec/bolt/cli_spec.rb

```diff
@@ -164,6 +165,69 @@ def stub_config(file_content = {})
     end
   end
 
+  context 'module' do
+    include BoltSpec::Project
+
+    let(:cli)     { Bolt::CLI.new(command + %W[--project #{project_path}]) }
+    let(:command) { ['module'] }
+
+    around(:each) do |example|
+      original = ENV['BOLT_MODULE_FEATURE']
+      ENV['BOLT_MODULE_FEATURE'] = 'true'
+
+      with_project do
+        example.run
+      end
+
+    ensure
+      ENV['BOLT_MODULE_FEATURE'] = original
+    end
+
+    before(:each) do
+      allow(Bolt::Project).to receive(:new).and_return(project)
+    end
+
+    it 'errors without BOLT_MODULE_FEATURE being set' do
+      ENV.delete('BOLT_MODULE_FEATURE')
+      expect { cli.parse }.to raise_error(Bolt::CLIError)
+    end
+
+    context 'without modules configured' do
+      it 'errors' do
+        expect { cli.parse }.to raise_error(Bolt::CLIError)
+      end
+    end
+
+    context 'with modules configured' do
+      let(:project_config) { { 'modules' => [] } }
+      let(:command)        { %w[module show] }
+
+      it 'does not error' do
+        expect { cli.parse }.not_to raise_error
+      end
+    end
+
+    describe 'install' do
```

Comment:
There's a set of tests for 'module install' starting at 231, but I've left them as-is for now since https://github.com/puppetlabs/bolt/pull/2172 makes some changes to them. Depending on which PR makes it in first, I'll rebase the other and update the tests accordingly.